qiskit/circuit/library/arithmetic/inverse_chebyshev.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Piecewise polynomial approximation to arcsin(1/x)."""
+
+from typing import Optional
+import numpy as np
+from numpy.polynomial.chebyshev import Chebyshev
+
+from qiskit.circuit import QuantumRegister
+from qiskit.circuit.library.blueprintcircuit import BlueprintCircuit
+from qiskit.circuit.exceptions import CircuitError
+
+from .piecewise_polynomial_pauli_rotations import PiecewisePolynomialPauliRotations
+from .integer_comparator import IntegerComparator
+
+# pylint: disable=invalid-name
+
+
+class InverseChebyshev(BlueprintCircuit):
+    r"""Piecewise Chebyshev approximation of the inverse function.
+
+    Building block of HHL. For a given constant :math:`C` and error tolerance :math:`\epsilon`,
+    this class implements a piecewise polynomial approximation on :math:`n` qubits to
+    :math:`arcsin(C/x)`, where :math:`x \in [a,2^n - 1]` and :math:`a` depends on :math:`\epsilon`.
+
+    The values of the parameters are calculated according to https://arxiv.org/abs/2009.04484.
```

Comment:
Can you add an examples section and change the reference to the style
```
.. according to [1].

References:

    [1] ...
```
Like in the other circuit library files.